manta-pay/abstract-protocol.tex

```diff
@@ -328,100 +350,140 @@ \subsection{Abstract Cryptographic Schemes}
 \end{definition}
 
-\begin{definition}[\Transfer{} Post]
-    A \TransferPost{} is the following tuple:
+To finish the transfer, the \SpendingKey{} for the $\Transfer.\ak : \ProofAuthorizingKey$ needs to sign the public side of the transaction. The public part of the transaction is the following post body:
+
+\begin{definition}[\Transfer{} Post Body]
+    A \TransferPostBody{} is the following tuple:
     \begin{align*}
-        \sources    &: \List(\Source) \\
-        \senders    &: \List(\SenderPost) \\
-        \receivers  &: \List(\ReceiverPost) \\
-        \sinks      &: \List(\Sink) \\
-        \pi         &: \NIZK.\Proof 
+        \ID        &: \Option(\AssetId) \\
+        \sources   &: \List(\Source) \\
+        \senders   &: \List(\SenderPost) \\
+        \receivers &: \List(\ReceiverPost) \\
+        \sinks     &: \List(\Sink) \\
+        \pi        &: \NIZK.\Proof 
     \end{align*}
-    A \TransferPost{}, $P$, is constructed by assembling the zero-knowledge proof of \Transfer{} validity from a known proving key $\pk : \NIZK.\ProvingKey$ and a given $T : \Transfer$:
-    \begin{align*}
-        x             &:= \Transfer.\public(T) \\
-        w             &:= \Transfer.\secret(T) \\
-        \Some(\pi)    &\sim \NIZK.\prove^\ValidTransfer_\pk(x, w) \\
-        P.\sources    &:= x.\sources \\
-        P.\senders    &:= x.\senders \\
-        P.\receivers  &:= x.\receivers \\
-        P.\sinks      &:= x.\sinks \\
-        P.\pi         &:= \pi
+    A \TransferPostBody{}, $B$, is constructed by assembling the zero-knowledge proof of \Transfer{} validity from a known proving key $\pk : \NIZK.\ProvingKey$ and a given $T : \Transfer$:
+   \begin{align*}
+        x            &:= \Transfer.\public(T) \\
+        w            &:= \Transfer.\secret(T) \\
+        \Some(\pi)   &\sim \NIZK.\prove^\ValidTransfer_\pk(x, w) \\
+        B.\ID        &:= x.\ID \\
+        B.\sources   &:= x.\sources \\
+        B.\senders   &:= x.\senders \\
+        B.\receivers &:= x.\receivers \\
+        B.\sinks     &:= x.\sinks \\
+        B.\pi        &:= \pi
     \end{align*}
     where $\Transfer.\public$ returns \SenderPost{s} for each \Sender{} in $T$ and \ReceiverPost{s} for each \Receiver{} in $T$, keeping \Source{s} and \Sink{s} as they are, and $\Transfer.\secret$ returns all the rest of $T$ which is not part of the output of $\Transfer.\public$.
 \end{definition}
 
+Now we can sign this body with $\sk_\alpha : \SpendingKey := \sk \cdot \alpha$ where the signature scheme has $\TransferPostBody$ as the $\SIG.\Message$ type and we use $\ak_\alpha$ as the verifying key:
+
+\begin{definition}[\Transfer{} Post]
+    A \TransferPost{} is the following tuple:
+    \begin{align*}
+        \sigma &: \Option(\SIG.\VerifyingKey \times \SIG.\Signature) \\
+        \body  &: \TransferPostBody
+    \end{align*}
+    Note that the $\sigma$ value is optional. This is inhabited whenever the number of \Sender{s} in a transaction is positive.
```

Comment:
How can there be a Transfer with no senders? Is it withdrawing from the PublicAssetPool?

Reply:
Transfer can by any shape with four integer free values: $[\textsf{sources}, \textsf{senders}, \textsf{receivers}, \textsf{sinks}]: \mathbb{N}^4$. For no $\textsf{senders}$ we can have a $\textsf{mint}$ which is $[1, 0, 1, 0]$.